cli/internal/terraform/terraform.go

```diff
@@ -597,7 +597,7 @@ func (c *Client) setLogLevel(logLevel LogLevel) error {
 }
 
 // StateMigration is a manual state migration that is not handled by Terraform due to missing features.
-// TODO(AB#3248): Remove this after we can assume that all existing clusters have been migrated.
+// TODO: Keep or remove? Currently not needed, but might be again in the future.
```

Comment:
I am leaning towards "keep" tbh. And making the function no-ops. Otherwise we will have to figure out how to add migrations once again soon. Same with the calls in the cmd pkg. 
Unless you have a plan to add a more refined migration path during the refactoring. And you expect that to finish during 2.11-pre.

Reply:
i have no strong opinion, but keeping the logic might make it easier for us. I can't foresee if we will need it again since it seemed like an edge case